Library/Homebrew/cmd/uses.rb

```diff
@@ -47,49 +47,48 @@ def uses
 
     uses = formulae.select do |f|
       used_formulae.all? do |ff|
-        if recursive
-          deps = f.recursive_dependencies do |dependent, dep|
-            if dep.recommended?
-              Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-            elsif dep.optional?
-              Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
-            elsif dep.build?
-              Dependency.prune unless includes.include?("build?")
+        begin
+          if recursive
+            deps = f.recursive_dependencies do |dependent, dep|
+              if dep.recommended?
+                Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
+              elsif dep.optional?
+                Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
+              elsif dep.build?
+                Dependency.prune unless includes.include?("build?")
+              end
             end
-
-            # If a tap isn't installed, we can't find the dependencies of one
-            # its formulae, and an exception will be thrown if we try.
-            if dep.is_a?(TapDependency) && !dep.tap.installed?
-              Dependency.keep_but_prune_recursive_deps
+            reqs = f.recursive_requirements do |dependent, req|
```

Comment:
After thinking about this for a bit, it's not such a great idea, because e.g. in the issue that this fixed, the issue was that a formula wasn't tapped, so a script that ran through all Homebrew/homebrew-* formulae likely wouldn't have run into it.

Is there a way to use analytics to see if the `rescue` block is ever hit? That way we could monitor it for a month or whatever and remove it if it's never hit.

If not, I'll add a warning for developers encouraging them to open an issue if they see it. After a while, if nobody reports anything, we could enable the warning for everybody, and then finally if nobody has still reported anything we could remove the `rescue`.